src/System.Management.Automation/engine/parser/tokenizer.cs

```diff
@@ -1717,6 +1730,9 @@ private void ScanToEndOfCommentLine(out bool sawBeginSig, out bool matchedRequir
 
         #region Object reuse
 
+        // A two-dimensional integer array reused for calculating string similarity.
+        private int[,] _beginTokenSimilarity2dArray;
```

Comment:
Since the size is fixed and appears to be used in almost all cases, does it make sense to initialize this in the constructor?

Reply:
There are scripts that doesn't have line comments at all and in those cases we don't want to allocate this shared 2-d array.